python/pyspark/ml/connect/readwrite.py

```diff
@@ -37,52 +38,99 @@ def sc(self) -> "SparkContext":
-        if isinstance(self._instance, JavaModel):
-            model = cast("JavaModel", self._instance)
+        if isinstance(instance, JavaModel):
+            model = cast("JavaModel", instance)
             params = serialize_ml_params(model, session.client)
             assert isinstance(model._java_obj, str)
             writer = pb2.MlCommand.Write(
                 obj_ref=pb2.ObjectRef(id=model._java_obj),
                 params=params,
                 path=path,
-                should_overwrite=self.shouldOverwrite,
-                options=self.optionMap,
+                should_overwrite=shouldOverwrite,
+                options=optionMap,
             )
-        else:
+            command = pb2.Command()
+            command.ml_command.write.CopyFrom(writer)
+            session.client.execute_command(command)
+
+        elif isinstance(instance, (JavaEstimator, JavaTransformer, JavaEvaluator)):
             operator: Union[JavaEstimator, JavaTransformer, JavaEvaluator]
-            if isinstance(self._instance, JavaEstimator):
+            if isinstance(instance, JavaEstimator):
                 ml_type = pb2.MlOperator.ESTIMATOR
-                operator = cast("JavaEstimator", self._instance)
-            elif isinstance(self._instance, JavaEvaluator):
+                operator = cast("JavaEstimator", instance)
+            elif isinstance(instance, JavaEvaluator):
                 ml_type = pb2.MlOperator.EVALUATOR
-                operator = cast("JavaEvaluator", self._instance)
-            elif isinstance(self._instance, JavaTransformer):
-                ml_type = pb2.MlOperator.TRANSFORMER
-                operator = cast("JavaTransformer", self._instance)
+                operator = cast("JavaEvaluator", instance)
             else:
-                raise NotImplementedError(f"Unsupported writing for {self._instance}")
+                ml_type = pb2.MlOperator.TRANSFORMER
+                operator = cast("JavaTransformer", instance)
 
             params = serialize_ml_params(operator, session.client)
             assert isinstance(operator._java_obj, str)
             writer = pb2.MlCommand.Write(
                 operator=pb2.MlOperator(name=operator._java_obj, uid=operator.uid, type=ml_type),
                 params=params,
                 path=path,
-                should_overwrite=self.shouldOverwrite,
-                options=self.optionMap,
+                should_overwrite=shouldOverwrite,
+                options=optionMap,
             )
-        command = pb2.Command()
-        command.ml_command.write.CopyFrom(writer)
-        session.client.execute_command(command)
+            command = pb2.Command()
+            command.ml_command.write.CopyFrom(writer)
+            session.client.execute_command(command)
+
+        elif isinstance(instance, (Pipeline, PipelineModel)):
+            from pyspark.ml.pipeline import PipelineSharedReadWrite
+
+            if shouldOverwrite:
```

Comment:
it is kind of tricky here, we don't have a method to remove the path in connect

Reply:
I'm wondering if we can have a DeleteFile command to do that?

Reply:
let me skip the support of `shouldOverwrite` first, we will need to find a better way